2_build_systems/2_build_systems.md

```diff
@@ -0,0 +1,240 @@
+## Compilers
+
+### What is a compiler?
+
+A compiler is a special program that translates a programming language's source code into machine code, bytecode or another programming language. The source code is typically written in a high-level, human-readable language such as C/C++.
+
+### Compiler Directives 
+
+Compiler directives refer to statement written in the source code of a program that lets the programmer instruct the compiler to perform a specific operation within the compilation phase itself.
+
+For example, the ```#include``` directive incorporates the contents of a header file to the compilation. Similarly, The ```#define``` directive is used to define a macro.
+
+### Different types of files during compilation
+
+<p align="center">
+  <img src="../../assets/images/Compilation_Stages.png" width="800" />
+</p>
+
+The compilation process transforms a human-readable code into a machine-readable format. For a programming language, it happens before a program starts executing to check the syntax and semantics of the code. The compilation process involves four steps: pre-processing, compiling, assembling, and linking then, we run the obtained executable file to get an output on the screen.
+
+* Source(.c):
+These files contain function definitions, and the entire program logics, these files are human readable and by convention their names ending with . c
+
+* Assembly(.s):
+Assembly Language mainly consists of mnemonic processor instructions or data and other statements or instructions.Flag -S eg. gcc -S foo.c
+
+* Object(.o):
+These files are produced as the output of the compiler. They consist of function definitions in binary form, but they are not executable by themselves and by convention their names end with .o.Flag -o eg. gcc -o foo.c
```

Comment:
![image](https://user-images.githubusercontent.com/64468582/220986648-33a67d8c-33e8-43de-a821-2df924a63790.png)
Can improve the readability using the proper formatting style like: 

These files are produced as the output of the compiler. They consist of function definitions in binary form, but they are not executable by themselves and by convention their names end with `.o`. 
`Flag: -o` 
`eg`:  gcc -o foo.c

Reply:
Resolved in latest commit